lib/turn.rb

```diff
@@ -0,0 +1,123 @@
+class Turn
+  attr_reader :player1, :player2, :spoils_of_war, :type
+
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    @spoils_of_war = []
+    #what is convention here?
+    if player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0)\
+      && (player1.deck.cards[2] == nil || player2.deck.cards[2] == nil)
+      @type = :war
+    elsif player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0)\
+      && player1.deck.rank_of_card_at(2) == player2.deck.rank_of_card_at(2)
+      @type = :mutually_assured_destruction
+    elsif player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0)
+      @type = :war
+    else
+      @type = :basic
+    end
+  end
+
+  def winner
+    if @type == :basic
+      self.basic_winner
+    elsif @type == :war
+      if player1.deck.cards.count < 3 || player2.deck.cards.count < 3
+        self.final_chance_war_winner
+      else
+        self.war_winner
+      end
+    else
+      "No Winner"
+    end
+  end
+
+  def basic_winner
+    if player1.deck.rank_of_card_at(0) > player2.deck.rank_of_card_at(0)
+      @player1
+    else
+      @player2
+    end
+  end
+
+  def war_winner
+    if player1.deck.rank_of_card_at(2) > player2.deck.rank_of_card_at(2)
+      @player1
+    else
+      @player2
+    end
+  end
+
+  def final_chance_war_winner
```

Comment:
When you add helper methods such as this one, `war_winner`, etc, keep in mind that these should be unit tested in isolation (for example, a test in `turn_spec.rb` which calls the `final_chance_war_winner` method directly and verifies its behavior).